fluent/src/types.js

```diff
@@ -28,6 +28,11 @@ export class FluentType {
    * This method can use `Intl` formatters memoized by the `MessageContext`
    * instance passed as an argument.
    *
+   * In most cases, valueOf returns a string, but it can be overriden
+   * and there are use cases, where the return type is not a string.
+   * An example if fluent-react where the Type returns a special React
+   * object.
```

Comment:
Maybe: `An example is fluent-react which implements a custom `FluentType` to represent React elements passed as arguments to format()`.